cub/test/catch2_test_device_reduce_large_offsets.cu

```diff
@@ -0,0 +1,247 @@
+
+#include "insert_nested_NVTX_range_guard.h"
+// above header needs to be included first
+
+#include <cub/device/device_reduce.cuh>
+#include <cub/thread/thread_operators.cuh>
+
+#include <thrust/iterator/counting_iterator.h>
+
+#include <cuda/std/__algorithm_>
+
+#include <cstdint>
+
+#include "catch2_test_device_reduce.cuh"
+#include "catch2_test_helper.h"
+#include "catch2_test_launch_helper.h"
+
+DECLARE_LAUNCH_WRAPPER(cub::DeviceReduce::Reduce, device_reduce);
+DECLARE_LAUNCH_WRAPPER(cub::DeviceReduce::Sum, device_sum);
+DECLARE_LAUNCH_WRAPPER(cub::DeviceReduce::Min, device_min);
+DECLARE_LAUNCH_WRAPPER(cub::DeviceReduce::ArgMin, device_arg_min);
+DECLARE_LAUNCH_WRAPPER(cub::DeviceReduce::Max, device_max);
+DECLARE_LAUNCH_WRAPPER(cub::DeviceReduce::ArgMax, device_arg_max);
+
+// %PARAM% TEST_LAUNCH lid 0:1:2
+
+// List of offset types to test
+using offset_types = c2h::type_list<std::int32_t, std::uint32_t, std::uint64_t>;
+
+_CCCL_HOST_DEVICE _CCCL_FORCEINLINE uint64_t
+get_segmented_guassian_sum(const uint64_t num_items, const uint64_t segment_size)
+{
+  const uint64_t sum_per_full_segment = (segment_size * (segment_size - 1)) / 2;
+  const uint64_t full_segments        = num_items / segment_size;
+  const uint64_t index_within_segment = num_items % segment_size;
+
+  const uint64_t sum_within_partial_segment = (index_within_segment * (index_within_segment - 1)) / 2;
+  const uint64_t sum_over_full_segments     = full_segments * sum_per_full_segment;
+  return sum_within_partial_segment + sum_over_full_segments;
+}
+
+template <typename ItemT>
+struct mod_op
+{
+  uint64_t segment_size;
+
+  _CCCL_HOST_DEVICE _CCCL_FORCEINLINE uint64_t operator()(const uint64_t index) const
+  {
+    return static_cast<ItemT>(index % segment_size);
+  }
+};
+
+CUB_TEST("Device reduce works with all device interfaces", "[reduce][device]", offset_types)
+{
+  using index_t  = uint64_t;
+  using offset_t = typename c2h::get<0, TestType>;
+
+  CAPTURE(c2h::type_name<offset_t>());
+
+  // Clamp 64-bit offset type problem sizes to just slightly larger than 2^32 items
+  const auto num_items_max_ull = ::cuda::std::clamp(
+    static_cast<std::size_t>(::cuda::std::numeric_limits<offset_t>::max()),
+    std::size_t{0},
+    ::cuda::std::numeric_limits<std::uint32_t>::max() + static_cast<std::size_t>(2000000ULL));
+  const offset_t num_items_max = static_cast<offset_t>(num_items_max_ull);
+  const offset_t num_items_min =
+    num_items_max_ull > 10000 ? static_cast<offset_t>(num_items_max_ull - 10000ULL) : offset_t{0};
+
+  // Generate the input sizes to test for
+  const offset_t num_items = GENERATE_COPY(
+    values(
+      {num_items_max, static_cast<offset_t>(num_items_max - 1), static_cast<offset_t>(1), static_cast<offset_t>(3)}),
+    take(2, random(num_items_min, num_items_max)));
+
+  // Input data
+  const auto index_it = thrust::make_counting_iterator(index_t{});
+
+  SECTION("reduce")
+  {
+    using op_t = cub::Sum;
```

Comment:
Suggestion: since we have a separate test section to test `Sum`, consider using a different reduction operation here.